Support tests in developer build.

severinstrobl opened the pull request:
When performing a developer build it should be possible to also run the tests, analogous to a regular install. This is useful when using Spack in a CI context where the build process also uses the Spack recipe.

severinstrobl committed:
Support tests in developer build.

When performing a developer build it should be possible to also run
the tests, analogous to a regular install.

adamjstewart commented:
In order to get the unit tests to pass, you need to run `spack commands --update-completion` and commit the changed file.

severinstrobl committed:
Bash completion for '--test' in dev-build.

Updated bash completion file for dev-build to support the newly added
argument '--test'.

severinstrobl commented:
Thanks for the reminder, I lost track of this one for a bit.

becker33 commented on lib/spack/spack/cmd/dev_build.py:
```diff
@@ -92,6 +102,13 @@ def dev_build(self, args):
+    if args.test == 'all':
+        tests = True
+    elif args.test == 'root':
+        tests = [spec.name for spec in abstract_specs]
```
```suggestion
        tests = [package.name]
```
This command can only take one spec at a time, and we've already checked that we only got one. We can just assign it directly.

severinstrobl replied:
Good point, thanks.

becker33 reviewed (changes requested):
Mostly LGTM, one minor request.

severinstrobl committed:
Integrating suggested changes for '--test' in dev-build

Co-authored-by: Greg Becker <becker33@llnl.gov>

severinstrobl committed:
Removed unused variable in dev-build.

The change in 11ab7af resulted in an unused variable.

severinstrobl commented:
At the time I created this request I didn't realize #17293 was addressing the same issue. With #17293 being merged, this pull request is obsolete, so I'll close it.
